ambry-store/src/main/java/com.github.ambry.store/BlobStoreCompactor.java

```diff
@@ -644,9 +644,9 @@ private boolean copyDataByIndexSegment(LogSegment logSegmentToCopy, IndexSegment
             try (BlobReadOptions options = srcIndex.getBlobReadInfo(indexEntry.getKey(),
                 EnumSet.allOf(StoreGetOptions.class))) {
               Offset offset = new Offset(indexSegmentStartOffset.getName(), options.getOffset());
-              IndexValue putValue =
-                  new IndexValue(options.getSize(), offset, options.getExpiresAtMs(), value.getOperationTimeInMs(),
-                      value.getServiceId(), value.getContainerId());
+              IndexValue putValue = new IndexValue(options.getMessageInfo().getSize(), offset,
```

Comment:
minor: Would this be more readable if you fetched the `MessageInfo` beforehand and put it into a variable?